### PR TITLE
WT-6965 Migrate search to use the new hs cursor implementation

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

**Review by hbokhari — APPROVED** [review]
LGTM

File: src/history/hs_cursor.c [hunk 1]
```diff
@@ -228,6 +220,7 @@ __hs_find_upd_int(WT_SESSION_IMPL *session, uint32_t btree_id, WT_ITEM *key,
 
     /* Allocate buffer for the history store value. */
     WT_ERR(__wt_scr_alloc(session, 0, &hs_value));
```
[the remark this answers is not in the document]
Comment (quchenhao) [reply src/history/hs_cursor.c]: Done

**Comment by quchenhao** [comment]
@bvpvamsikrishna There is a system issue for macos on evergreen. Other than the that, the tests look good.

**Review by bvpvamsikrishna — APPROVED** [review]
lgtm

**Pull request merged**